### PR TITLE
Pack

### PR DESCRIPTION
**Pull request by stefanpenner**
separate collaborators for different packing algorithms (pack/index.js)
------------------------------------------------------------------

- [x] bin
- [x] fixed dimension
- [x] shelf first
- [x] improve plugability for bins. (likely via list-view delegate API)

list-view
-----------

- [x] integrate with new bins
- [x] demo for new shelf-first packing
- [x] entangle bins and
- [x] lots of tests
- [x] move padding calculations to the bins
- [x] sometimes large array changes in grid mode, result in rendering glitches (need to isolate the scenario)

- [ ] bug: changing rowHeights when near the end of the list.
- [x] bug: replacing entire set causes mis-calculations (I believe I have a reproduction: https://github.com/emberjs/list-view/commit/f791c349451b470a3c4c550dfb4ceddf4159fb9e)

-- would be nice
- [ ] visibleStartingIndex needs to also memoize top on the entry + likely binary sort across the search space to get something close to `log2(n)+1` rather then being `O(n)`
- [ ] improve API ergonomics (they really suck now)
- [ ] extract pack.js into its own micro-lib (because that is what the cool kids do)
- [x] support ember > 1.9

### CODE REVIEW

**Comment by SebastianSzturo** [comment]
Excited

**Comment by kumavis** [comment]
this has been a long time coming! nice work

**Comment by stefanpenner** [comment]
list-view position syncing isn't working correctly when scrolled down and large chunks of content are removed.

File: packages/list-view/lib/list_view_mixin.js [hunk 1]
```diff
@@ -742,7 +684,7 @@ export default Ember.Mixin.create({
 
     delta = numberOfChildViewsNeeded - numberOfChildViews;
 
-    if (delta === 0) {
+       if (delta === 0) {
```
Comment (annismckenzie) [inline packages/list-view/lib/list_view_mixin.js]: Did you really want to add this whitespace here? :)

**Comment by stefanpenner** [comment]
superseded by #204

**Pull request closed without merging**